### PR TITLE
Issue 20: Youtube video player block

### PR DESCRIPTION
**Pull request by tmathern**
Fix https://github.com/hlxsites/caesars-careers/issues/20

## 🔗 Test URLs:

- Before: https://main--caesars-careers--hlxsites.hlx.page/drafts/mathern/video
- After: https://issue-20-youtube-video--caesars-careers--hlxsites.hlx.page/drafts/mathern/video

## 📝 Description:

- Adds a Youtube video player block

### CODE REVIEW

File: careers/blocks/video/video.css [hunk 1]
```diff
@@ -0,0 +1,44 @@
+.video.block {
+  position: relative;
+  width: 100%;
+  height: 560px;
+  border: 1px solid red;
```
Comment (arumsey) [inline careers/blocks/video/video.css:5]: Test border can be removed and hide any potential overflow.

```suggestion
  overflow: hidden;
```

File: careers/blocks/video/video.css [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+.video.block .video-player iframe {
+  width: 100%;
+  height: 560px;
+  aspect-ratio: 16 / 9;
+}
+
+.video.block .video-poster > div {
+  display: flex;
+  flex-flow: column;
+}
```
Comment (arumsey) [inline careers/blocks/video/video.css:26]: I don't think this is needed anymore.

```suggestion

```

File: careers/blocks/video/video.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  if (!window.YT) {
+    pendingPlayers.push({ id: videoId, element: videoPlayer.firstElementChild });
+  } else {
+    loadYouTubePlayer(videoPlayer.firstElementChild, videoId);
+  }
+  if (!window.onYouTubeIframeAPIReady) {
+    // onYouTubeIframeAPIReady will load the video after the script is loaded
+    window.onYouTubeIframeAPIReady = () => {
+      pendingPlayers.forEach(({ id, element }) => loadYouTubePlayer(element, id));
+    };
```
Comment (arumsey) [inline careers/blocks/video/video.js:71]: @tmathern Does commenting out this block help your score? Calling `loadYouTubePlayer` could be deferred until the `element` scrolls into view.

**Comment by tmathern** [comment]
https://github.com/hlxsites/caesars-careers/pull/38

**Review by amol-anand — APPROVED** [review]

**Pull request merged**